src/encoder/serde.rs

```diff
@@ -496,6 +496,17 @@ impl Serialize for TimeStamp {
     }
 }
 
+impl Serialize for RegExp {
```

Comment:
This follows the pattern of the other serializers defined here. Since the serialize implementation for Bson implements serializing into extjson, we piggyback off that to serialize a RegExp.